dashboard/app/controllers/lessons_controller.rb

```diff
@@ -18,16 +18,18 @@ def show
   def edit
     @lesson_data = {
       id: @lesson.id,
-      name: @lesson.name,
-      overview: @lesson.overview,
-      studentOverview: @lesson.student_overview,
-      assessment: @lesson.assessment,
-      unplugged: @lesson.unplugged,
-      lockable: @lesson.lockable,
-      creativeCommonsLicense: @lesson.creative_commons_license,
-      purpose: @lesson.purpose,
-      preparation: @lesson.preparation,
-      announcements: @lesson.announcements
+      lesson: {
```

Comment:
Seems strange to use "lesson" here if there will be things about the lesson which aren't editable which might also be returned. Like the ID, here. Or do you think we won't have any of those?

Reply:
I can see the argument for `editableData`. for example, we might include the localized name here, which is part of "lesson", but is not "editable". 

Where I'm stuck with this is in the next PR, where I try to move us onto a serializer: https://github.com/code-dot-org/code-dot-org/pull/36894 In that situation, I kind of want all of the different places we serialize to output in the same format, rather than the `show` page using `lesson` (or no "envelope") and the `edit` page using `editableData`. This might not be a good goal, though. Any thoughts? 

For now I'll change this back to `editableData` and we can discuss this in terms of what we'll do in the next PR.

Reply:
Low confidence level here on all of this, but here's where I'm at:

I'd prefer having `show` and `edit` not be consistent, and be specialized for their specific pages. If we were making very general set of APIs that would be consumed in a lot of places on the site, I think we would want that consistency, but it seems like right now they support one specific page each. With this format it feels like for whichever page I'm working on, I can see the data the server gives me and understand what it means without having to look at the server code or docs.

I'd also be ok with making `show` and `edit` consistent - either both use the same envelope key or no envelope. If we were doing this, I think validating and raising errors on unknown parameters should be a hard requirement, since we'd lose the self-documenting `editableData`. We'd also want to make it easy to figure out what's editable and what's not from looking at the code or comments.

Whatever we choose, I feel like it's pretty easy to change it before it's fully launched, and we'll get more of a sense of whether we like it as we develop with them a bit more. So I hope it doesn't feel like we're locked into what we decide now.